client/src/client.rs

```diff
@@ -50,21 +49,77 @@ impl<R> Default for QueryResponseHandler<R> {
     }
 }
 
+/// `Result` with [`ClientQueryError`] as an error
+pub type QueryHandlerResult<T> = core::result::Result<T, ClientQueryError>;
+
 impl<R> ResponseHandler for QueryResponseHandler<R>
 where
     R: Query + Into<QueryBox> + Debug,
     <R::Output as TryFrom<Value>>::Error: Into<eyre::Error>,
 {
-    type Output = PaginatedQueryOutput<R>;
+    type Output = QueryHandlerResult<ClientQueryOutput<R>>;
+
+    fn handle(self, resp: Response<Vec<u8>>) -> Self::Output {
+        // Separate-compilation friendly response handling
+        fn _handle_query_response_base(
```

Comment:
This isn't necessary. We are using LTO for release builds, and prefer inlining when possible. Iroha consumes a lot of RAM anyway, might as well make it faster. This pattern would make sense in an embedded context, but in our release mode it is very unlikely to reduce the code size. 

Thank you for the effort, though.

Reply:
This function is used in a lot of places, and I thought it can optimize compilation time **a bit** and it should reduce code size a bit. Also, it is a **client library**, not Iroha internal stuff. It is not supposed to be used from within Iroha. So, it might make sense to make this little optimization.

Reply:
I get where you're coming from, and it might make a huge difference in some cases. However, as per Donald Knuth

[Premature optimisation is the root of all evil](https://stackify.com/premature-optimization-evil/)